### ISSUE TITLE
Merging in-use Local Authorities fails

### ISSUE THREAD

**Issue by mark-dce** [curationexperts/alexandria-v2#480]
**STEPS TO REPRODUCE**
1. Find or create a new Local Subject
2. Make sure one or more repository objects reference the Local Subject
3. Test this by trying to delete the Local Subject, you should get an error saying "Record 'XXX' cannot be deleted because it is referenced by #N other records."

For example, try http://sandbox.curationexperts.com/catalog/0d0976b4-b3e6-4c96-8534-bf33a9d72bf8

4. Create a new Local Subject to merge the existing Local Subject into
5. Go back to the initial Local Subject, and choose "Merge this record" from the tool bar
6. Enter the new Local Subject in the merge box
7. Click "yes" on the merge confirmation
8. You should get a message that "A background job has been queued to merge the records."

**EXPECTED BEHAVIOR**
a. The source record is marked as deleted (i.e. it shouldn't show up as an option in local authority fields any more
b. The object records pointing to the source (from step 1) authority URI should be updated to point to the merge target URI (from step 4)
c. The merge target URI should no longer be deletable and instead report that it has records referencing it.

**ACTUAL RESULTS**
a. The original authority record continues to be visible
b. no objects appear to be updated to point at the new authority record
c. the new authority record can still be deleted

**Comment by jcoyne** [comment]
Works for me.

Initial state
<img width="659" alt="screen shot 2016-01-07 at 12 07 17 pm" src="https://cloud.githubusercontent.com/assets/92044/12178532/9307e4ac-b537-11e5-9429-29d7fd598409.png">

Select the record to merge
<img width="806" alt="screen shot 2016-01-07 at 12 07 45 pm" src="https://cloud.githubusercontent.com/assets/92044/12178535/970b648e-b537-11e5-9680-4965ec2104a0.png">

Flash appears
<img width="436" alt="screen shot 2016-01-07 at 12 07 59 pm" src="https://cloud.githubusercontent.com/assets/92044/12178539/9f542bc6-b537-11e5-8959-15f32d08a203.png">

Final state.  "George" has been removed.
<img width="974" alt="screen shot 2016-01-07 at 12 08 25 pm" src="https://cloud.githubusercontent.com/assets/92044/12178542/a274b618-b537-11e5-9bbe-aa9fabc6b754.png">

**Comment by jcoyne** [comment]
I was also able to reproduce my successful test on sandbox

**Comment by mark-dce** [comment]
@jcoyne  
Cool - can you test it with subjects instead of persons, and specifically with this subject: http://sandbox.curationexperts.com/catalog/0d0976b4-b3e6-4c96-8534-bf33a9d72bf8

That's the one that failed for me that I gave the steps to reproduce above.

**Comment by jcoyne** [comment]
Yes, I  now see that it is not working for subjects.

**Comment by jcoyne** [comment]
It looks like when you tested it it failed because Marmotta was down.  Now it is failing because it can't validate the namespace:

```
n
    ActiveFedora::RecordInvalid
Error
    Validation failed: value `http://127.0.0.1:8080/fedora/rest/prod/a6/3e/59/ba/a63e59ba-e088-4090-8d50-ffa116bbe637' for `lc_subject' property is not a term in a controlled vocabulary lcsh, lcnames, tgm, aat, local

    /opt/alex2/shared/bundle/ruby/2.2.0/gems/active-fedora-9.7.0/lib/active_fedora/validations.rb:56:in `save!'
    /opt/alex2/releases/20160107163347/app/services/merge_records_service.rb:79:in `block in update_records_with_new_reference'
    /opt/alex2/releases/20160107163347/app/services/merge_records_service.rb:67:in `each'
    /opt/alex2/releases/20160107163347/app/services/merge_records_service.rb:67:in `update_records_with_new_reference'
    /opt/alex2/releases/20160107163347/app/services/merge_records_service.rb:24:in `run'
    /opt/alex2/releases/20160107163347/app/jobs/merge_records_job.rb:7:in `perform'
    /opt/alex2/shared/bundle/ruby/2.2.0/gems/activejob-4.2.5/lib/active_job/execution.rb:32:in `block in perform_now'
    /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.5/lib/active_support/callbacks.rb:117:in `call'
    /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.5/lib/active_support/callbacks.rb:117:in `call'
    /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.5/lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
    /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.5/lib/active_support/callbacks.rb:505:in `call'
    /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.5/lib/active_support/callbacks.rb:505:in `call'
    /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.5/lib/active_support/callbacks.rb:498:in `block (2 levels) in around'
```

**Comment by jcoyne** [comment]
It appears that only occurs when the term to be merged is referenced by another object. I suspect it's causing an error when it's trying to set a string without first casting to an RDF::URI.  We probably need to add a check for that here: https://github.com/no-reply/linked_vocabs/blob/master/lib/linked_vocabs/validators/property_validator.rb#L7

**Comment by jcoyne** [comment]
Fixed by e680527e9fb090b75b0209d5d1e7b73c31b12334
